common/src/main/java/com/microsoft/identity/common/internal/broker/BrokerValidator.java

```diff
@@ -78,13 +78,12 @@ public BrokerValidator(final Context context) {
      * Verifies that the installed broker package's signing certificate hash matches the known
      * release certificate hash.
      *
-     * If signature hash verification fails, this will throw a Client exception, with its error message being a list of hashes.
+     * If signature hash verification fails, this will throw a Client exception containing the cause of error, which could contain a list of mismatch hashes.
      *
      * @param brokerPackageName The broker package to inspect.
-     * @return True if the certificate hash is known. False otherwise.
+     * @return SignatureHash of brokerPackageName, if the verification succeeds..
      */
-    public boolean verifySignatureAndThrowIfHashVerificationFailed(final String brokerPackageName) throws ClientException {
-        final String methodName = ":verifySignatureAndThrowIfHashVerificationFailed";
+    public String verifySignatureAndThrow(final String brokerPackageName) throws ClientException {
```

Comment:
>verifySignatureAndThrow [](start = 18, length = 23)

I wish we log telemetry in this method instead of anywhere , if the signature validation fails...

Reply:
Once telemetry is wired into common, that'll be doable. As of now we'll have to pass it to ad-accounts :(